webui/react/src/contexts/AppContexts.tsx

```diff
@@ -54,49 +45,32 @@ const AppContexts: React.FC = () => {
     } catch (e) {}
   }, [ canceler, setAgents, setOverview ]);
 
-  const fetchAll = useCallback((): void => {
-    triggerCommandsRequest({});
-    triggerNotebooksRequest({});
-    triggerShellsRequest({});
-    triggerTensorboardsRequest({});
-    triggerActiveExperimentsRequest({ states: activeRunStates });
-  }, [
-    triggerCommandsRequest,
-    triggerNotebooksRequest,
-    triggerShellsRequest,
-    triggerTensorboardsRequest,
-    triggerActiveExperimentsRequest,
-  ]);
+  const fetchCommands = useFetchCommands(canceler);
+  const fetchNotebooks = useFetchNotebooks(canceler);
+  const fetchShells = useFetchShells(canceler);
+  const fetchTensorboards = useFetchTensorboards(canceler);
 
   const fetchUsers = useCallback((): void => {
     triggerUsersRequest({ url: '/users' });
   }, [ triggerUsersRequest ]);
 
+  usePolling(fetchActiveExperiments);
   usePolling(fetchAgents);
-  usePolling(fetchAll);
+  usePolling(fetchCommands);
```

Comment:
wdy think about keeping these under a single fetchTasks, or similar call so there is no need for extra usePolling calls

Reply:
I think that smaller/dedicated functions are better for re-usability in the future.

Reply:
there might be a miscommunication here 🤔  agree having separate fetch functions is good I'm not saying we shouldn't keep those. I'm thinking about passing them onto usePolling once (like how it was). IIRC each react hook we added onto a component added an additional re-render to the page. @hkang1 would know better than me here.

Reply:
agree with @hamidzr here. Though the individual fetches are not related to each other directly, I think there are a couple of benefits for keeping them together. The re-render is reduced with fewer `usePolling` hooks. IMO, it's generally a good thing to just have less `setTimeouts` in general, as it can complicate the residual cleanup step. Speaking of that, the `usePolling` returns a `stopPolling` function, which can be called when the component / page unmounts. Something like this:
```
  const stopPolling = usePolling(fetchX);

  useEffect(() => {
    return () => {
      stopPolling();
      canceler.abort();
    };
  }, [ canceler, stopPolling ]);
```

Reply:
Are you sure `usePolling` is directly causing re-renders? I might be wrong but I think re-renders are caused by (eg) `setCommands`, which updates the state in the context, and causes a re-render where the state is listened.

Added the fix btw, because surely having less timeouts is better.